test/Models/TerriaSpec.ts

```diff
@@ -865,6 +867,225 @@ describe("Terria", function() {
+        url: magdaRecordDerefencedToWmsUrl,
+        recordId: "magda-record-id-dereferenced-to-wms",
+        id: "another-test-magda-record"
+      };
+
+      const mapServerModelWithError = {
+        type: "esri-mapServer-group",
+        name: "A map server with error",
+        url: mapServerWithErrorUrl,
+        id: "a-test-server-with-error"
+      };
+
+      const theItemsIds = [
+        "a-test-server-group/0",
+        "a-test-magda-record/0",
+        "another-test-magda-record"
+      ];
+
+      let loadMapItems: any = undefined;
+      beforeEach(function() {
+        const realLoadWithXhr = loadWithXhr.load;
+        spyOn(loadWithXhr, "load").and.callFake(function(...args: any[]) {
+          let url = args[0];
+
+          if (url.match("mapServerSimpleGroup")) {
+            if (url.indexOf("MapServer?f=json") !== -1) {
+              args[0] =
+                "test/Terria/applyInitData/MapServer/mapServerSimpleGroup.json";
+            } else if (url.indexOf("MapServer/0?f=json") !== -1) {
+              args[0] = "test/Terria/applyInitData/MapServer/0.json";
+            } else {
+              args[0] = "test/Terria/applyInitData/empty.json";
+            }
+          } else if (url.match("mapServerWithError")) {
+            if (url.indexOf("MapServer?f=json") !== -1) {
+              args[0] =
+                "test/Terria/applyInitData/MapServer/mapServerWithError.json";
+            } else if (url.indexOf("MapServer/0?f=json") !== -1) {
+              args[0] = "test/Terria/applyInitData/MapServer/0.json";
+            } else {
+              args[0] = "test/Terria/applyInitData/empty.json";
+            }
+          } else if (
+            url.match("magda-record-id-dereferenced-to-feature-server-group")
+          ) {
+            args[0] =
+              "test/Terria/applyInitData/MagdaReference/group_record.json";
+          } else if (url.match("magda-record-id-dereferenced-to-wms")) {
+            args[0] =
+              "test/Terria/applyInitData/MagdaReference/wms_record.json";
+          } else if (url.match("services2.arcgis.com")) {
+            if (url.indexOf("FeatureServer?f=json") !== -1) {
+              args[0] =
+                "test/Terria/applyInitData/FeatureServer/esri_feature_server.json";
+            } else if (url.indexOf("FeatureServer/0?f=json") !== -1) {
+              args[0] = "test/Terria/applyInitData/FeatureServer/0.json";
+            } else {
+              args[0] = "test/Terria/applyInitData/empty.json";
+            }
+          } else if (url.match("mapprod1.environment.nsw.gov.au")) {
+            if (url.indexOf("request=GetCapabilities") !== -1) {
+              args[0] = "test/Terria/applyInitData/WmsServer/capabilities.xml";
+            } else {
+              args[0] = "test/Terria/applyInitData/empty.json";
+            }
+          }
+
+          const result = realLoadWithXhr(...args);
+          return result;
+        });
```

Comment:
Is there a reason why this doesn't follow the normal pattern we use for mocking xhr calls?

For example

https://github.com/TerriaJS/terriajs/blob/6f04346ca07899a69c05f3ffb07d8f8caf97482c/test/Table/TableStyleSpec.ts#L25-L41